### PR TITLE
Add table aws_auditmanager_framework. Closes #400

### LINKED ISSUE #400: Add table aws_auditmanager_framework
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_auditmanager_framework []

PRETEST: tests/aws_auditmanager_framework

TEST: tests/aws_auditmanager_framework
Running terraform
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "aws auditmanager list-assessment-frameworks --framework-type Standard --output json --profile integration-tests --region us-east-1 > /private/var/folders/d8/gsp6f_cn3359xnt_gs3cts_80000gn/T/tests/aws_auditmanager_framework/terraform/test/output.json"]
null_resource.named_test_resource: Creation complete after 4s [id=4903950200015524174]
data.local_file.input: Reading...
data.local_file.input: Read complete after 0s [id=96702f4a2f21a9b7b00102263c1a4b595c06058d]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

arn = "arn:aws:auditmanager:us-east-1::assessmentFramework/21060514-60c0-3fbb-ad3b-0b4af6dcca7c"
aws_account = "013122550996"
aws_partition = "aws"
aws_region = "us-east-1"
id = "21060514-60c0-3fbb-ad3b-0b4af6dcca7c"
name = "AWS Well Architected Framework"
type = "Standard"

Running SQL query: test-get-query.sql
[
  {
    "arn": "arn:aws:auditmanager:us-east-1::assessmentFramework/21060514-60c0-3fbb-ad3b-0b4af6dcca7c",
    "id": "21060514-60c0-3fbb-ad3b-0b4af6dcca7c",
    "name": "AWS Well Architected Framework",
    "partition": "aws",
    "type": "Standard"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:auditmanager:us-east-1::assessmentFramework/21060514-60c0-3fbb-ad3b-0b4af6dcca7c",
    "id": "21060514-60c0-3fbb-ad3b-0b4af6dcca7c",
    "name": "AWS Well Architected Framework",
    "partition": "aws",
    "type": "Standard"
  }
]
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:auditmanager:us-east-1::assessmentFramework/21060514-60c0-3fbb-ad3b-0b4af6dcca7c"
    ],
    "title": "AWS Well Architected Framework"
  }
]
✔ PASSED

POSTTEST: tests/aws_auditmanager_framework

TEARDOWN: tests/aws_auditmanager_framework

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  name,
  arn,
  id,
  type
from
  aws_auditmanager_framework
where
  type = 'Custom';
+------+------------------------------------------------------------------------------------------------------+--------------------------------------+--------+
| name | arn                                                                                                  | id                                   | type   |
+------+------------------------------------------------------------------------------------------------------+--------------------------------------+--------+
| test | arn:aws:auditmanager:us-east-1:013122550996:assessmentFramework/47ec5126-54ba-425a-80a2-da4c5f39c80b | 47ec5126-54ba-425a-80a2-da4c5f39c80b | Custom |
+------+------------------------------------------------------------------------------------------------------+--------------------------------------+--------+
```
</details>

### CODE REVIEW

**Comment by bigdatasourav** [comment]
used arnToAkas as transform.EnsureStringArray is not returning correct value.

File: aws-test/tests/aws_audit_manager_framework/test-turbot-expected.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[
+	{
+		"akas": ["{{ output.arn.value }}"],
+		"title": "{{ output.name.value }}"
+	}
+]
```
Comment (Subhajit97) [inline aws-test/tests/aws_audit_manager_framework/test-turbot-expected.json:6]: @bigdatasourav Seems like the file indented with 4 spaces. Can you please check your vscode settings?

File: aws-test/tests/aws_audit_manager_framework/variables.tf [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+locals {
+  path = "${path.cwd}/output.json"
+}
+
+resource "null_resource" "named_test_resource" {
+  provisioner "local-exec" {
+    command = "aws auditmanager list-assessment-frameworks --framework-type Standard --output json --profile ${var.aws_profile} --region ${data.aws_region.primary.name} > ${local.path}"
+  }
+}
```
Comment (Subhajit97) [inline aws-test/tests/aws_audit_manager_framework/variables.tf:51]: @bigdatasourav Are we using the predefined framework, or we need to create one?

File: aws/table_aws_audit_manager_framework.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+			{
+				Name:        "logo",
+				Description: "The logo associated with the framework.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "type",
+				Description: "The framework type, such as standard or custom.",
+				Type:        proto.ColumnType_STRING,
+			},
```
Comment (Subhajit97) [inline aws/table_aws_audit_manager_framework.go:102]: Add this column after `id`

File: aws/table_aws_audit_manager_framework.go [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAuditManagerFramework(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	auditType := plugin.GetMatrixItem(ctx)[matrixKeyAudit].(string)
+	plugin.Logger(ctx).Debug("listAuditManagerFrameworks", "AuditType", auditType, "REGION", region)
+	// Create Session
```
Comment (Subhajit97) [inline aws/table_aws_audit_manager_framework.go:164]: Add new line before comments

File: aws/utils.go [hunk 5]
```diff
@@ -145,4 +145,8 @@ func getLastPathElement(path string) string {
 
 func lastPathElement(_ context.Context, d *transform.TransformData) (interface{}, error) {
 	return getLastPathElement(types.SafeString(d.Value)), nil
-}
\ No newline at end of file
+}
+
+func convertSafeString(_ context.Context, d *transform.TransformData) (interface{}, error) {
+	return types.SafeString(d.Value), nil
+}
```
Comment (Subhajit97) [inline aws/utils.go:152]: Why we need this `*string` to `string` conversion here?
You can use `transform.EnsureStringArray` function directly

File: aws/multi_region.go [hunk 6]
```diff
@@ -73,3 +83,51 @@ func BuildWafRegionList(_ context.Context, connection *plugin.Connection) []map[
 		{matrixKeyRegion: GetDefaultRegion()},
 	}
 }
+
+// BuildAuditRegionList :: return a list of matrix items for AWS audit resources, one per region specified in the connection config
+func BuildAuditRegionList(ctx context.Context, connection *plugin.Connection) []map[string]interface{} {
```
Comment (Subhajit97) [inline aws/multi_region.go:88]: @bigdatasourav Can you please put a comment, why we need this function?

File: aws/table_aws_audit_manager_framework.go [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAuditManagerFramework,
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Arn").Transform(convertSafeString).Transform(transform.EnsureStringArray),
```
Comment (Subhajit97) [inline aws/table_aws_audit_manager_framework.go:127]:
```suggestion
				Transform:   transform.FromField("Arn").Transform(transform.EnsureStringArray),
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@bigdatasourav There are few changes required. Please see comments, thanks!

File: aws-test/tests/aws_audit_manager_framework/variables.tf (hunk 2 above)
Comment (bigdatasourav) [reply aws-test/tests/aws_audit_manager_framework/variables.tf:51]: dont need to create for test.

File: aws/utils.go (hunk 5 above)
Comment (bigdatasourav) [reply aws/utils.go:152]: arnToakas used as discuss

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@bigdatasourav Can you please use multi-region setup and verify the GET call once?
Seems like `framework-id` is same in every region.
```
> select name, id from aws_shaktiman.aws_audit_manager_framework where id = 'c9bed0e0-ac88-35c7-bc64-61ea073dce0c'
Error: get call returned 2 results - the key column is not globally unique
```

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Comment by cbruno10** [comment]
In TF, even though Audit Manager resources are not available, it looks like they're using the format `aws_auditmanager_*` (https://github.com/hashicorp/terraform-provider-aws/blob/ceb4c185a26c1635b6f3d26f3adce1be301a3667/.hashibot.hcl#L141), so we should follow the same format. 

Can you please rename the table and files to use this naming structure instead?

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**